### PR TITLE
Determine legend width in PostScript

### PR DESCRIPTION
**Pull request by PaulWessel**
Let most legends (1 column) be typeset in _PostScript_ where text lengths can be computed exactly.
See #1790 for background.

### CODE REVIEW

File: doc/examples/ex03/ex03.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ gmt begin ex03
 		gmt subplot set 0,0 -A"Input Power"
 		gmt plot spectrum.xpower -JX-?l/?l -Bxa1f3p -Bya1f3p+l"Power (mGal@+2@+km)" -Gred -ST0.07i -R1/1000/0.1/10000 -Ey+p0.5p -lShip
 		gmt plot spectrum.ypower -Gblue -Sc0.07i -Ey+p0.5p -lSatellite
-		gmt legend -DjBL+w1.2i+o0.25i -F+gwhite+pthicker --FONT_ANNOT_PRIMARY=14p,Helvetica-Bold
+		gmt legend -DjBL+o0.25i -F+gwhite+pthicker --FONT_ANNOT_PRIMARY=14p,Helvetica-Bold
```
Comment (seisman) [inline doc/examples/ex03/ex03.sh]: Also update ex03.bat and ex07.bat to keep them in sync.

**Comment by PaulWessel** [comment]
I think this PR is good to merge.  Chekcs pass, the plots looks much better, etc. OK with @seisman and @joa-quim ?

File: doc/rst/source/legend.rst [hunk 2]
```diff
@@ -80,13 +80,17 @@ Auto-legends
...
-different settings you must call **legend** explicitly instead.
+*specfile* from individual entries for each command.  The **-l** option takes an optional
+label and allows optional modifiers **+d**, **+g**, **+n**, **+h**, **+j**, **+l**, and **+v**
+that follow the corresponding uppercase legend codes discussed above.  In addition,
+there is **+f** to set header (H) or label (L) font,  **+s** to set the symbol size (or line length)
+to use for the current entry, and **+x** to scale all symbols uniformly (see **-S** above).
+Some defaults are hardwired: We draw a white rectangular panel with
+a 1 point black outline offset from the justification point (**+j**) by 0.2 cm.  To use
+different settings you must call **legend** explicitly before :doc:`end` does so
+implicitly instead.  Note: With an explicit call to **legend** you can also use **-S** to
```
[the remark this answers is not in the document]
Comment (PaulWessel) [reply doc/rst/source/legend.rst]: OK done

File: doc/rst/source/postscriptlight.rst [hunk 3]
```diff
@@ -323,6 +323,11 @@ affect the current state of parameters such as line and fill attributes.
     PDF transparency, set *rgb*\ [3] to a value between 0 (opaque) and 1
     (fully transparent).
 
+**long PSL_setcurrentpoin** (**struct PSL_CTRL** *\*P*, **double**
```
[the remark this answers is not in the document]
Comment (PaulWessel) [reply doc/rst/source/postscriptlight.rst]: OK done

**Pull request merged**